sdk/textanalytics/ai-text-analytics/src/textAnalysisClient.ts

```diff
@@ -0,0 +1,490 @@
+    options: TextAnalysisClientOptions = {}
+  ) {
+    const { defaultCountryHint = "us", defaultLanguage = "en", ...pipelineOptions } = options;
+    this.defaultCountryHint = defaultCountryHint;
+    this.defaultLanguage = defaultLanguage;
+
+    const internalPipelineOptions: InternalPipelineOptions = {
+      ...pipelineOptions,
+      ...{
+        loggingOptions: {
+          logger: logger.info,
+          additionalAllowedHeaderNames: ["x-ms-correlation-request-id", "x-ms-request-id"],
+        },
+      },
+    };
+
+    this._client = new GeneratedClient(endpointUrl, internalPipelineOptions);
+
+    const authPolicy = isTokenCredential(credential)
+      ? bearerTokenAuthenticationPolicy({ credential, scopes: DEFAULT_COGNITIVE_SCOPE })
+      : textAnalyticsAzureKeyCredentialPolicy(credential);
+
+    this._client.pipeline.addPolicy(authPolicy);
+    this._tracing = createTracingClient({
+      packageName: "@azure/ai-text-analytics",
+      packageVersion: SDK_VERSION,
+      namespace: "Microsoft.CognitiveServices",
+    });
+  }
+
+  /**
+   * Runs a predictive model to determine the language that the passed-in
+   * input strings are written in, and returns, for each one, the detected
+   * language as well as a score indicating the model's confidence that the
+   * inferred language is correct.  Scores close to 1 indicate high certainty in
+   * the result.  120 languages are supported.
+   *
+   * See {@link https://docs.microsoft.com//azure/cognitive-services/language-service/concepts/data-limits}
+   * for data limits.
+   *
+   * ### Examples
+   *
+   * #### Language detection
+   *
+   * ```js
+   * const documents = [<input strings>];
+   * const countryHint = "us";
+   * const results = await client.analyze("LanguageDetection", documents, countryHint);
+   *
+   * for (let i = 0; i < results.length; i++) {
+   *   const result = results[i];
+   *   if (result.error) {
+   *     // a document has an error instead of results
+   *   } else {
+   *     const { name, confidenceScore, iso6391Name } = result.primaryLanguage;
+   *   }
+   * }
+   * ```
+   *
+   * See {@link https://docs.microsoft.com//azure/cognitive-services/language-service/language-detection/overview}
+   * for more information on language detection.
+   *
+   * @param actionName - the name of the action to be performed on the input
+   *   documents, see ${@link AnalyzeActionName}
+   * @param documents - the input documents to be analyzed
+   * @param options - optional action parameters and settings for the operation
+   *
+   * @returns an array of results where each element contains the primary language
+   *   for the corresponding input document.
+   */
+  public async analyze<ActionName extends "LanguageDetection">(
+    actionName: ActionName,
+    documents: LanguageDetectionInput[],
+    options?: AnalyzeActionParameters<ActionName> & TextAnalysisOperationOptions
+  ): Promise<AnalyzeResult<ActionName>>;
```

Comment:
I avoided doing this because for some reason it provides a worse VSCode UX experience:

https://user-images.githubusercontent.com/6074665/164274923-3887e0ea-6235-4450-9fbb-a9cebbce52ed.mp4

Reply:
That's wild that it leaves the first overload open when you start typing `"S`, definitely not what I expected.

Reply:
Can you file a bug on TS for this? 👍